### ISSUE TITLE
Native application equivalent in set of documents vs set of software programs

### ISSUE THREAD

**Issue by jamieherrera** [w3c/wcag2ict#308]
In reviewing the Guidance on Applying WCAG 2.2 to Non-Web Information and Communications Technologies (WCAG2ICT) and its predecessor, it seems unclear what a native application on a phone should be. 

In my work on native apps, in al companies I've been with, a native app screen is considered roughly equivalent to a web page; a set of webpages often is a website. A set of app screens makes up a single native application. The "set" in this case is shard common content such as a navigation bar or header. 

I don't even know how one might apply Bypass Blocks to a set of native applications, which is the current interpretation in some spaces.  

1. Should a native application be considered a "set of documents" or a "set of software programs" for the purpose of this document? 
2. Could you please add more native app examples to the document, such as where these phrases are used?
3. Is the Mobile Accessibility Task Force partnering on this document?

**Comment by patrickhlauke** [comment]
see https://github.com/w3c/wcag2ict/issues/261

based on the outcomes of that discussion (which I don't necessarily agree with, but it is what it is), I'd say in answer to your first question: a single native application (unless it's a "suite" of applications or similar) won't count as a "set of documents" nor "set of software programs").

for some SCs, this makes sense and simplifies things (heck, bypass blocks, or multiple ways, make little sense for a more app-like experience, even when it comes to just web applications). while in other cases (such as 2.4.8 Location or 3.2.3 Consistent Navigation / 3.2.4 Consistent Identification) it's weird that it won't be applicable.

**Comment by jamieherrera** [comment]
Thanks Patrick, I read through #261 and #54, which do outline some of the complexities of "non-web" content and how changing definitions might force certain SC to apply in ways that are not intended. 
But at some point within the new version of Guidance on Applying WCAG 2.2 to Non-Web Information and Communications Technologies (WCAG2ICT), there should be some reference to native mobile applications, as separate from the document's discussion of mobile phones or mobile browsers. As in, either they are explicitly not applicable, in which case punt the reader to [Mobile Accessibility: How WCAG 2.0 and Other W3C/WAI Guidelines Apply to Mobile ](https://www.w3.org/TR/mobile-accessibility-mapping/)(or maybe there's an updated version more recent than the [2018 Editor's Draft](https://w3c.github.io/Mobile-A11y-TF-Note/) in play for 2.2 also)? Or, they are applicable in specific circumstances, in which case examples and use cases would be warranted. 

Question 3 seems silly if the Mobile Accessibility document is intended as a sibling document to this "Non-web" document, but since the relationship is not clear, I should probably still ask it.

**Comment by maryjom** [comment]
DRAFT ANSWER, not yet reviewed by the WCAG2ICT Task Force. 

Native mobile applications are covered by the WCAG2ICT Note as "non-web software". For the rest of your comments each of the paragraphs and numbered questions are repeated below with the pertinent reply after each part.

> In my work on native apps, in all companies I've been with, a native app screen is considered roughly equivalent to a web page; a set of webpages often is a website. A set of app screens makes up a single native application. The "set" in this case is shared common content such as a navigation bar or header.

The task force does not consider a native app screen or view as a web page. Nor does it consider a native application, as a whole, as a website. It is a (singular) non-web software application. You can see some of our reasoning in the [thread in Issue 225](https://github.com/w3c/wcag2ict/issues/225#issuecomment-1783604904). Additionally, WCAG conformance claims are on a per-page basis where non-web software makes conformance claims for the entire software  application.

> I don't even know how one might apply Bypass Blocks to a set of native applications, which is the current interpretation in some spaces.

The navigation problem that exists in the Web does not exist in native software. Bypass Blocks is one where (if applied) non-web software would likely automatically satisfy the requirement. Web content has the problem that one cannot quickly navigate to the main part of the content - you have to Tab past every menu and navigation link whenever there is a left or top navigation on a website. For native desktop application, one simply uses a keystroke to toggle between the menu and the main part of the application.  For mobile applications, a user simply taps where they want to start interacting.  If navigating sequentially using a screen reader, mobile applications also have much smaller top level menus, meaning the burden of getting past them to get to the main part of the application is minimal - especially when they employ the use of the hamburger menu.

The task force is unaware of any official interpretation that requires bypass blocks or any of the other "sets of software" requirements for native software applications - mobile or desktop. 

NOTE: There are no standards that require any of the "set of software" requirements for native software of any kind - including mobile. As examples, neither the European standard EN 301 549 (including derivative works in Australia, Canada, India, etc.) nor the U.S. Revised 508 standards apply any of the Success Criteria that are scoped to "sets of software programs" to non-web software - including mobile apps. In the future, those standards may choose rewrite some of those WCAG success criteria to more clearly apply and scope them with language that is more meaningful for native software. The WCAG2ICT Task Force is not at liberty to do so, as our work statement says we are to steer clear of developing requirements, substantively changing the requirements, or creating development or test techniques, etc.

Regarding your 3 questions:

>1. Should a native application be considered a "set of documents" or a "set of software programs" for the purpose of this document?
The simple answer is no. It would be a very unusual case for a native application to be considered a "[set of software programs](https://w3c.github.io/wcag2ict/#set-of-software-programs)" per the key term definition in the document. We do not plan to look for examples that would meet the definition, as it is likely there are none. Jonathan Avila posited in a related Issue #225 that Microsoft Office is an example of a set of software applications, though that is a native desktop software example.

>2. Could you please add more native app examples to the document, such as where these phrases are used?
Providing specific examples would make this document brittle over time as technology changes. Even the term "mobile" has changed from being used for smartphones to also include smart watches and tablets. We currently do not know of any examples of a set of software programs for mobile.

>3. Is the Mobile Accessibility Task Force partnering on this document?
No, the Mobile Accessibility Task Force is not partnering with the WCAG2ICT Task Force. They have been inactive as a Task Force for some time. This week a new chair was announced for the group, so activity may reignite. As part of the W3C Note process, this document goes out for review, on a regular basis, to all of the Task Forces and the entire Accessibility Guidelines. Content is sent for review for small groups of newly added or changed content, as well as each time a public draft is released.
 
Of particular note, it wasn't and still isn't within the allowable scope of the WCAG2ICT Task Force to substantively change the WCAG language to apply. That work is better done by a standards body that is working specifically on accessibility requirements for native software. To my knowledge, the European standards organization ETSI will be looking into this as they include new WCAG 2.2 requirements in an update to the EN 301 549.

There are other related issues #225, #261, and #54 in which similar questions were asked, as well as more information  provided about the considerations that WCAG2ICT had when coming up with "sets of software" and why it was a tricky topic for all WCAG criteria that applied to sets of web pages and websites.

**Comment by pday1** [comment]
Possible minor tweaks to text: 
Deletions shown in ~~strikethrough~~
Additions/changes shown in **bold**

Bypass Blocks is one where (if applied) non-web software would likely automatically satisfy the requirement. Web content has the problem that one cannot quickly navigate to the main part of the content - you have to ~~Tab~~ **tab** past every menu and navigation link whenever there is a left or top navigation on a website. For **a** native desktop application, one simply uses a keystroke to toggle between the menu and the main part of the application.  For mobile applications, a user simply taps where they want to start interacting.  If navigating sequentially using a screen reader, mobile applications also have much smaller top level menus, meaning the burden of getting past them to get to the main part of the application is minimal - especially when they employ the use of the hamburger menu.

**Comment by maryjom** [comment]
Not making assumptions, but trying to make a point that native apps are much less likely to have the navigation issues solved by Bypass Blocks. That SC is currently not even being applied by 508 or the EN 301 549 to native applications.

**Comment by mraccess77** [comment]
A few people have mentioned simply tapping after the content - but I'm not so sure this is that easy as if you are blind you don't know exactly where to tap and you may miss something and some folks are using VoiceOver with input from a braille display device.  Clearly there are relevant commands like the rotor which can support navigation by heading of course - but I don't think we can assume everyone can just tap past the content.

**Comment by maryjom** [comment]
FINAL WCAG2ICT TF ANSWER:

Native mobile applications are covered by the WCAG2ICT Note as "non-web software". For the rest of your comments each of the paragraphs and numbered questions are repeated below with the pertinent reply after each part.

> In my work on native apps, in all companies I've been with, a native app screen is considered roughly equivalent to a web page; a set of webpages often is a website. A set of app screens makes up a single native application. The "set" in this case is shared common content such as a navigation bar or header.

The task force does not consider a native app screen or view as a web page. Nor does it consider a native application, as a whole, as a website. It is a (singular) non-web software application. You can see some of our reasoning in the [thread in Issue 225](https://github.com/w3c/wcag2ict/issues/225#issuecomment-1783604904). Additionally, WCAG conformance claims are on a per-page basis where non-web software makes conformance claims for the entire software  application.

> I don't even know how one might apply Bypass Blocks to a set of native applications, which is the current interpretation in some spaces.

It wasn't and still isn't within the allowable scope of the WCAG2ICT Task Force to substantively change the WCAG language to apply more clearly to non-web documents or software. Nor can we create development or test techniques. That work is better done by a standards body that is working specifically on accessibility requirements for native software. To our knowledge, the European standards organization ETSI may be looking into this specific issue as they work on an update to the EN 301 549 to include WCAG 2.2. Where WCAG2ICT notes that some of the "set of" requirements are considered a "best practice", ETSI may decide to create a requirement with appropriate scope and language.

Regarding your 3 questions:

>1. Should a native application be considered a "set of documents" or a "set of software programs" for the purpose of this document?

The simple answer is no. It would be a very unusual case for a native application to be considered a "[set of software programs](https://w3c.github.io/wcag2ict/#set-of-software-programs)" per the key term definition in the document. We do not plan to look for examples that would meet the definition, as it is likely there are none. Jonathan Avila posited in a related Issue #225 that Microsoft Office is an example of a set of software applications, though that is a native desktop software example.

>2. Could you please add more native app examples to the document, such as where these phrases are used?

Providing specific examples would make this document brittle over time as technology changes. Even the term "mobile" has changed from being used for smartphones to also include smart watches and tablets. We currently do not know of any examples of a set of software programs for mobile.

>3. Is the Mobile Accessibility Task Force partnering on this document?

No, the Mobile Accessibility Task Force is not partnering with the WCAG2ICT Task Force. They have been inactive as a Task Force for some time. Recently a new chair was announced for the group, so activity may reignite. As part of the W3C Note process, this document goes out for review, on a regular basis, to all of the Task Forces and the entire Accessibility Guidelines Working Group. Content is sent out for review for small groups of newly added or changed content, as well as each time a public draft is released.

There are other related issues #225, #261, and #54 in which similar questions were asked. These provide more information about the considerations that WCAG2ICT had when coming up with "sets of software" and why it was a tricky topic for all WCAG criteria that applied to sets of web pages and websites.

**Comment by maryjom** [comment]
Closing as answered. 

See also the related issue #145 regarding "sets of" requirements where the task force will be adding statements to look to the normative standards for how WCAG is actually being applied. Since WCAG2ICT is not normative, and is simply guidance, the developers of standards such as the EN 301 549 were active in the WCAG2ICT task force and used the interpretation in the WCAG2ICT Note to guide how the requirements were (or weren't) applied to non-web documents and software.

**Comment by jamieherrera** [comment]
Thank you.
